### PR TITLE
add a proposed interface to crypto in Git

### PR DESCRIPTION
**Pull request by maxtaco**
- Made a trivial but very broad change, which is to bring the `TLFVisibility` enum into `keybase1.` rather than `chat1.`, since we'll need it, and it has little to do with chat
  - That small change made a huge disruption

### CODE REVIEW

File: go/git/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package git
```
Comment (maxtaco) [inline go/git/interfaces.go]: @patrickxb this is really what I wanted to do, but everything else happened when I needed a `TeamIDWithVisibility`

**Comment by maxtaco** [comment]
cc: @keybase/react-hackers, this small protocol change required a bunch of JS changes.

**Comment by maxtaco** [comment]
@cjb do these JS changes look ok?  They passed flow, and I'm hoping that's good enough.

**Comment by maxtaco** [comment]
@chrisnojima? Bueller?

**Comment by keybase-ci-visdiff** [comment]
The commits 414ccb626ab9...4f53b7ea5848 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504564657313-414ccb626ab9-4f53b7ea5848-linux/Markdown-emoji-414ccb626ab9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504564657313-414ccb626ab9-4f53b7ea5848-linux/Markdown-emoji-4f53b7ea5848.png)  
   Markdown-emoji comparison
</summary>

**Comment by maxtaco** [comment]
Failure is unrelated KBFS CI flake.

**Pull request merged**